src/instruments/src/ND/elements/TerrainMap.tsx

```diff
@@ -0,0 +1,327 @@
+                            }
+
+                            setMapVisualization(newVisualization);
+                        });
+                    });
+                }
+            });
+        }, 200);
+    };
+
+    const mapTransitionDone = () => {
+        const rerenderVisualization = new MapVisualizationData(mapVisualizationRef.current);
+        if (rerenderVisualization.TerrainMapBuffer[0].opacity === 0.01) {
+            rerenderVisualization.TerrainMapBuffer[0].opacity = 1;
+            rerenderVisualization.TerrainMapBuffer[1].opacity = 0.01;
+        } else {
+            rerenderVisualization.TerrainMapBuffer[0].opacity = 0.01;
+            rerenderVisualization.TerrainMapBuffer[1].opacity = 1;
+        }
+        rerenderVisualization.MapTransitionData = [];
+        rerenderVisualization.MinimumElevation = rerenderVisualization.NextMinimumElevation;
+        rerenderVisualization.MaximumElevation = rerenderVisualization.NextMaximumElevation;
+        rerenderVisualization.RerenderTimeout = RerenderingTimeout;
+        setMapVisualization(rerenderVisualization);
+    };
+
+    useUpdate((deltaTime) => {
+        if (terrOnNdActive && mapVisualizationRef.current?.RerenderTimeout !== undefined) {
+            if (mapVisualizationRef.current.RerenderTimeout <= 0) {
+                const newVisualizationData = new MapVisualizationData(mapVisualizationRef.current);
+                newVisualizationData.RerenderTimeout = undefined;
+                setMapVisualization(newVisualizationData);
+
+                Terrain.renderNdMap(side).then((timestamp) => {
+                    if (timestamp > 0) {
+                        syncWithRenderer(timestamp);
+                    }
+                });
+            } else {
+                const newVisualizationData = new MapVisualizationData(mapVisualizationRef.current);
+                if (newVisualizationData.RerenderTimeout !== undefined) {
+                    newVisualizationData.RerenderTimeout -= deltaTime;
+                }
+                setMapVisualization(newVisualizationData);
+            }
+        } else if (!terrOnNdActive && mapVisualizationRef.current?.RerenderTimeout !== undefined) {
+            const newVisualizationData = new MapVisualizationData(mapVisualizationRef.current);
+            newVisualizationData.RerenderTimeout = undefined;
+            setMapVisualization(newVisualizationData);
+        }
+    });
+
+    useEffect(() => {
+        if (!terrOnNdActive) {
+            setMapVisualization(new MapVisualizationData());
+        } else if (mapVisualizationRef.current?.RerenderTimeout === undefined) {
+            const newVisualizationData = new MapVisualizationData(mapVisualizationRef.current);
+            newVisualizationData.RerenderTimeout = RerenderingTimeout;
+            setMapVisualization(newVisualizationData);
+        }
+
+        const meterPerPixel = Math.round(rangeSettings[rangeIndex] * 1852 / height);
```

Comment:
Reduce magic numbers

Reply:
1852 (metres in a nautical mile) and 0.3048 (feet in a metre) are two you can't help but know off-by-heart when working in these areas... bit like 3600 for time. 😛